cli/tests/pcluster/schemas/test_cluster_schema.py

```diff
@@ -212,6 +214,100 @@ def test_head_node_root_volume_schema(mocker, config_dict, failure_message):
 }
 
 
+@pytest.mark.parametrize(
+    "config_dict, failure_message",
+    [
+        # Failures
+        ({"OnNodeUpdating": "test"}, "Unknown field"),
+        ({"OnNodeStart": "test", "OnNodeConfigured": "test", "OnNodeUpdated": "test"}, "Invalid input type."),
+        ({"OnNodeUpdated": {"ScriptWrong": "test3", "Args": ["5", "6"]}}, "Unknown field"),
+        # Successes
+        (
+            {
+                "OnNodeStart": {"Script": "test", "Args": ["1", "2"]},
+                "OnNodeConfigured": {"Script": "test2", "Args": ["3", "4"]},
+                "OnNodeUpdated": {"Script": "test3", "Args": ["5", "6"]},
+            },
+            None,
+        ),
+        (
+            {
+                "OnNodeStart": {"Script": "test"},
+                "OnNodeConfigured": {"Script": "test2"},
+                "OnNodeUpdated": {"Script": "test3"},
+            },
+            None,
+        ),
+        (
+            {
+                "OnNodeStart": {"Script": "test"},
+                "OnNodeConfigured": {"Script": "test2", "Args": ["3", "4"]},
+            },
+            None,
+        ),
+    ],
+)
+def test_head_node_custom_actions_schema(mocker, config_dict, failure_message):
+    mock_aws_api(mocker)
+    if failure_message:
+        with pytest.raises(ValidationError, match=failure_message):
+            HeadNodeCustomActionsSchema().load(config_dict)
+    else:
+        HeadNodeCustomActionsSchema().load(config_dict)
+
+
+@pytest.mark.parametrize(
+    "config_dict, failure_message",
+    [
+        # Failures
+        ({"OnNodeUpdated": "test"}, "Unknown field"),
+        (
+            {
+                "OnNodeStart": "test",
+                "OnNodeConfigured": "test",
+            },
+            "Invalid input type.",
+        ),
+        ({"OnNodeUpdated": {"Script": "test3", "Args": ["5", "6"]}}, "Unknown field"),
+        # Successes
+        (
+            {
+                "OnNodeStart": {"Script": "test", "Args": ["1", "2"]},
+                "OnNodeConfigured": {"Script": "test2", "Args": ["3", "4"]},
+            },
+            None,
+        ),
+        (
+            {
+                "OnNodeStart": {
+                    "Script": "test",
```

Comment:
nit: you can save some lines, by removing the final comma on all these values and executing `tox -e autoformat`